Empire/Empire.cpp

```diff
@@ -142,7 +142,7 @@ namespace {
             if (excess_here < EPSILON)
                 continue;
             // Transfer excess to stockpile
-            new_contributions += excess_here * stockpile_transfer_efficiency;
+            new_contributions = std::max(new_contributions + excess_here, stockpile_limit);
```

Comment:
is this revised calc right, particularly in light of how it gets used in the following line?
```return starting_stockpile + new_contributions - stockpile_used;```

  For one, it seems to me it should be an std::min operation rather than max, and second, suppose that ```starting_stockpile``` was already at the ```stockpile_limit```, and that none is used, and ```new_contributions``` cap out at the stockpile limit then that return value could be up to twice the ```stockpile_limit``` 

I think I'd suggest letting this line 145 be simply something like 
``` new_contributions += excess_here```

and then putting the limit in the return line
```return std::min(starting_stockpile + new_contributions - stockpile_used, stockpile_limit);```

Reply:
Current version is (w/o comments and log line):
````
            new_contributions += excess_here;
        }
        if (new_contributions > stockpile_limit)
            new_contributions = stockpile_limit;
        return starting_stockpile + new_contributions - stockpile_used;
````
Doing an `std::min` at the end with `stockpile_used` isn't needed, as it's supposed to limit the inputs, not the inputs after the withdrawals are deducted, and the above `if` lines limit the contributions to `stockpile_limit`.